internal/controllers/pod_controller.go

```diff
@@ -132,6 +118,24 @@ func (r *PodReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+func (r *PodReconciler) checkPod(ctx context.Context, pod *corev1.Pod) (string, error) {
+	var ns corev1.Namespace
+	if err := r.Get(ctx, client.ObjectKey{Name: pod.Namespace}, &ns); err != nil {
+		return "", err
+	}
+
+	result, err := r.Validator.ValidatePod(ctx, pod, &ns)
+	if err != nil {
+		return "", err
+	}
+
+	resultLabelValue := labelForValidationResult(result)
+	if resultLabelValue == "" {
```

Comment:
This `if` probably is redundant bacause it is equal with `return resultLabelValue, nil` for `resultLabelValue == ""`

Reply:
Hmm you're right :)

Reply:
Now we decided keep this solution